internal/drivers/manager.go

```diff
@@ -98,6 +98,9 @@ func (m *Manager) GetInstanceByStageID(ctx context.Context, poolName, stage stri
 	}
 
 	pool := m.poolMap[poolName]
+	if pool == nil {
+		return nil, fmt.Errorf("GetInstanceByStageID: pool name %q not found", poolName)
```

Comment:
Let's log the error message, also add stage_runtime_id like L107.

Reply:
updated